src/main/java/org/broadinstitute/hellbender/engine/FeatureManager.java

```diff
@@ -375,6 +380,22 @@ public boolean isEmpty() {
         return dataSource.iterator();
     }
 
+    /**
+     * As above, but takes an optional list of intervals to examine.
+     * @param featureDescriptor FeatureInput to scan
+     * @param intervals The userIntervals to examine (may be null)
+     * @param <T> Feature type
+     * @return An iterator over the Features
+     */
+    public <T extends Feature> Iterator<T> getFeatureIterator( final FeatureInput<T> featureDescriptor,
+                                                               final List<SimpleInterval> intervals ) {
+        final FeatureDataSource<T> dataSource = lookupDataSource(featureDescriptor);
+        if ( intervals != null ) {
+            dataSource.setIntervalsForTraversal(intervals);
+        }
+        return dataSource.iterator();
```

Comment:
As currently written, this method has the side effect of permanently setting the traversal intervals on the underlying data source, which will affect future calls to `getFeatureIterator()` as well as other methods. I recommend that you get rid of this side effect by clearing the traversal intervals on the data source after you obtain the iterator. Eg.,

```
if ( intervals != null ) {
    dataSource.setIntervalsForTraversal(intervals);
}
final Iterator<T> iter = dataSource.iterator();
dataSource.setIntervalsForTraversal(null);
return iter;
```

Would be less clunky if there were an `iterator(List<SimpleInterval>)` method we could use here, but there isn't, unfortunately.

Reply:
I opted to always setIntervalsForTraversal before getting the iterator, and always set them to null afterwards.  Fewer assumptions about the state of the source.  OK?